src/main/java/org/folio/dew/controller/BulkEditController.java

```diff
@@ -307,4 +352,50 @@ private int getIdentifierIndex(JobCommand jobCommand) {
       throw new NonSupportedEntityException(format("Non-supported entity type: %s", jobCommand.getEntityType()));
     }
   }
+
+  private void processBulkEditUpdateUploadCSV(Path uploadedPath, JobCommand jobCommand, MultipartFile file) throws IOException, CsvRequiredFieldEmptyException, CsvDataTypeMismatchException {
+    if (jobCommand.getEntityType() == ITEM) {
+      processBulkEditUpdateUploadCSV(uploadedPath, file, ItemFormat.class);
+    } else {
+      processBulkEditUpdateUploadCSV(uploadedPath, file, UserFormat.class);
+    }
+  }
+
+  private <T> void processBulkEditUpdateUploadCSV(Path uploadedPath, MultipartFile file, Class<T> clazz) throws IOException, CsvRequiredFieldEmptyException, CsvDataTypeMismatchException {
+    List<T> csvLines = getDifferenceBetweenInitialAndEditedRecordsCSV(file.getInputStream(), clazz);
+    if (csvLines.isEmpty()) { // If no records changed, just write column headers.
+      Files.write(uploadedPath, clazz == ItemFormat.class ? ItemFormat.getItemColumnHeaders().getBytes() : UserFormat.getUserColumnHeaders().getBytes());
+    } else {
+      CsvHelper.saveRecordsToCsv(csvLines, clazz, uploadedPath.toFile().getAbsolutePath());
+    }
+  }
+
+  private <T> List<T> getDifferenceBetweenInitialAndEditedRecordsCSV(InputStream edited, Class<T> clazz) {
+    return new CsvToBeanBuilder<T>(new InputStreamReader(edited))
+      .withType(clazz)
+      .withSkipLines(1)
+      .build()
+      .parse()
+      .stream()
+      .filter(recordFormat -> {
+        if (clazz == UserFormat.class) {
+          return applyUserFilter((UserFormat) recordFormat);
+        } else {
+          return applyItemFilter((ItemFormat) recordFormat);
+        }
+      })
+      .collect(Collectors.toList());
+  }
+
+  private boolean applyUserFilter(UserFormat userFormat) {
+    var userFromDB = userClient.getUserById(userFormat.getId());
```

Comment:
Better to use some another variable as value return from another module, but not from database.